packages/runtime/src/helpers/analysis.ts

```diff
@@ -0,0 +1,103 @@
+import fs from 'fs'
+
+import { extractExportedConstValue, UnsupportedValueError } from 'next/dist/build/analysis/extract-const-value'
+import { parseModule } from 'next/dist/build/analysis/parse-module'
+import { relative } from 'pathe'
+
+export interface ApiStandardConfig {
+  type?: never
+  runtime?: 'nodejs' | 'experimental-edge'
+  schedule?: never
+}
+
+export interface ApiScheduledConfig {
+  type: 'experimental-scheduled'
+  runtime?: 'nodejs'
+  schedule: string
+}
+
+export interface ApiBackgroundConfig {
+  type: 'experimental-background'
+  runtime?: 'nodejs'
+  schedule?: never
+}
+
+export type ApiConfig = ApiStandardConfig | ApiScheduledConfig | ApiBackgroundConfig
+
+export const validateConfigValue = (config: ApiConfig, apiFilePath: string): config is ApiConfig => {
+  if (config.type === 'experimental-scheduled') {
+    if (!config.schedule) {
+      console.error(
+        `Invalid config value in ${relative(
+          process.cwd(),
+          apiFilePath,
+        )}: schedule is required when type is "experimental-scheduled"`,
+      )
+      return false
+    }
+    if ((config as ApiConfig).runtime === 'experimental-edge') {
```

Comment:
The cast isn't necessary

```suggestion
    if (config.runtime === 'experimental-edge') {
```

Reply:
Gotcha. For some reason it doesn't giving me red squigglies in my editor (even restarted the TS lang server in VS Code) when I remove the cast. But during compilation I see the error. `npm ERR! src/helpers/analysis.ts(44,9): error TS2367: This condition will always return 'false' since the types '"nodejs"' and '"experimental-edge"' have no overlap.`